lib/logstash/inputs/http.rb

```diff
@@ -333,4 +335,11 @@ def error_details(e, trace = false)
     error_details
   end
 
+  def initialize_codec(codec_name)
+    codec_klass = LogStash::Plugin.lookup("codec", codec_name)
+    return codec_klass.new unless defined?(::LogStash::Plugins::Contextualizer)
+
+    ::LogStash::Plugins::Contextualizer.initialize_plugin(execution_context, codec_klass)
```

Comment:
Minor style preference, but up to you: since there's only one instruction after the early return, the full formed conditional `if` can help readability.

```suggestion
    if defined?(::LogStash::Plugins::Contextualizer)
      ::LogStash::Plugins::Contextualizer.initialize_plugin(execution_context, codec_klass)
    else
      codec_klass.new 
    end
```